modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManager.java

```diff
@@ -220,25 +214,15 @@ else if (matches > 1)
         return extensionRegistry.getConfigurationProvider(configurationProviderName);
     }
 
-    private boolean attemptToCreateImplicitConfiguration(ExtensionModel extensionModel, MuleEvent muleEvent)
+    private void createImplicitConfiguration(ExtensionModel extensionModel, MuleEvent muleEvent)
     {
         synchronized (extensionModel)
         {
             //check that another thread didn't beat us to create the instance
-            if (!extensionRegistry.getConfigurationProviders(extensionModel).isEmpty())
+            if (extensionRegistry.getConfigurationProviders(extensionModel).isEmpty())
```

Comment:
if `getConfigurationProviders` is empty, then you need to register a new connection provider. Otherwise, at least one has already been registered, so do nothing.